tests/hooks/test_postgres_hook.py

```diff
@@ -39,7 +39,10 @@ def setUp(self):
             schema='schema'
         )
 
-        self.db_hook = PostgresHook()
+        class UnitTestPostgresHook(PostgresHook):
+            conn_name_attr = 'test_conn_id'
```

Comment:
I'm not sure this change actually tests the change though

Reply:
Once I had tried a few things, I ended up with this test-driven development path:

1. Make the change above
2. Run `tests/hooks/test_postgres_hook.py` and get this error (also in the PR description):
``python
AttributeError: 'UnitTestPostgresHook' object has no attribute 'postgres_conn_id'
```
3. Make the change to `airflow/hooks/postgres_hook.py` to call the `conn_name_attr` rather than directly calling `postgres_conn_id`
4. See the tests pass

I think I what you called out sounds right-on: there is a `conn_name_attr` used for connecting to any database using the `DbApiHook`, which is partly why we went ahead and change this attribute when we sub-classed `PostgresHook`.

Anything else you all would need to see to make this change?

Thanks for considering!

Reply:
Something like this:

```python
    @mock.patch('airflow.hooks.postgres_hook.psycopg2.connect')
    def test_get_conn_non_default_id(self, mock_connect):
        self.db_hook.test_conn_id = "non_default"
        self.db_hook.get_conn()
        mock_connect.assert_called_once_with(user='login', password='password', host='host',
                                             dbname='schema', port=None)
        self.db_hook.get_connection.assert_called_once_with("non_default")
```

That way it is a little bit more explicit (and easier to track down if it breaks).